mls-rs/src/group/mod.rs

```diff
@@ -602,6 +604,155 @@ where
         self.context().epoch
     }
 
+    #[cfg_attr(not(mls_build_async), maybe_async::must_be_sync)]
+    async fn hpke_encrypt_to_recipient_with_generic_context(
+        &self,
+        recipient_index: u32,
+        context_info: &[u8],
+        associated_data: Option<&[u8]>,
+        plaintext: &[u8],
+    ) -> Result<HpkeCiphertext, MlsError> {
+        let member_leaf_node = self
+            .group_state()
+            .public_tree
+            .get_leaf_node(LeafIndex(recipient_index))?;
+        let member_public_key = &member_leaf_node.public_key;
+        let hpke_ciphertext = self
+            .cipher_suite_provider
+            .hpke_seal(member_public_key, context_info, associated_data, plaintext)
+            .await
+            .map_err(|e| MlsError::CryptoProviderError(e.into_any_error()))?;
+        Ok(hpke_ciphertext)
+    }
+
+    /// HPKE encrypts a message to the member at the specified `recipient_index` in the group.
+    ///
+    /// Takes `context_info`, `associated_data`, and `plaintext`.
+    /// Returns `ciphertext` and `kem_output` inside `HpkeCiphertext`.
+    ///
+    /// WARNING: The message sender is not authenticated.
+    #[cfg(all(feature = "non_domain_separated_hpke_encrypt_decrypt", feature = "ffi"))]
+    #[cfg_attr(
+        not(mls_build_async),
+        maybe_async::must_be_sync,
+        safer_ffi_gen::safer_ffi_gen_ignore
+    )]
```

Comment:
This is horrible on our part. FWIW we hope to get rid of safer-ffi and rely instead on mls-rs wrappers using uniffi.

Reply:
No problem! It was easy enough to figure out the required cfgs. But that'll be nice in the future :smiley: